### PR TITLE
Legacy Fatigue

### PR DESCRIPTION
**Pull request by PabstMirror**
Rewrote BIS's Legacy Fatigue from A3 Samples

I'm not sure if this is something we want.  ~~The new "Stamina" isn't great but it's not completely awful anymore.~~ With 1.54 you can run forever @ 14kph, so it actually is pretty shit and not at all realistic.

Really only possible because of all the added commands in 1.54.

Being scripted we can have full control over over drain and recovery and offer more options changeable by mission makers, but it's also scripted instead of simple config changes.

### CODE REVIEW

**Comment by jonpas** [comment]
Maybe a lead setting to toggle it on and off?

**Comment by PabstMirror** [comment]
With the changes to `CfgMovesFatigue` the script part still needs to run, but it should be possible to still behave pretty much like vanilla 1.54 stamina.

File: addons/LegacyFatigue/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ace_x
+==========
+
+## Maintainers
+
+The people responsible for merging changes to this component or answering potential questions.
+
+- [x](https://github.com/x)
```
Comment (bux578) [inline addons/LegacyFatigue/README.md]: whoa ... who's that guy?

**Comment by bux578** [comment]
Looking forwared to this. This should be released shortly after 1.54 :)

File: addons/LegacyFatigue/XEH_postInit.sqf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+if (!hasInterface) exitWith {};
+
+
+["SettingsInitialized", {
+
+    //Start on ACE_player
+    if (!isNull ACE_player) then {
+        [ACE_player] call FUNC(legacyFatigue);
+        ACE_LOGINFO("Starting fatigue on %1", _newPlayer);
+    };
```
Comment (commy2) [inline addons/LegacyFatigue/XEH_postInit.sqf]: Should be covered by the event below already.

File: addons/LegacyFatigue/config.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+#include "ACE_Settings.hpp"
+#include "CfgAnimStamina.hpp"
+#include "CfgEventHandlers.hpp"
+#include "CfgVehicles.hpp"
+
+class RscInGameUI {
+    class RscStaminaBar {
+        onLoad = "[""onLoad"",_this,""RscStaminaBar"",'IGUI'] call 	(uinamespace getvariable 'BIS_fnc_initDisplay');  uiNamespace setVariable ['test5', _this select 0];";
```
Comment (commy2) [inline addons/LegacyFatigue/config.cpp]: test5 
reminder to remove / rename

File: addons/LegacyFatigue/functions/fnc_legacyFatigue.sqf [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+[{
+    BEGIN_COUNTER(all);
+    params ["_args", "_pfID"];
+    _args params ["_unit", "_lastFastRun", "_lastFullRun", "_scriptedChangePerSecond", "_scriptedChangeDone", "_lastPos", "_lastStamina", "_exhaustionEnd"];
+
+    if ((isNull _unit) || {!alive _unit} || {_unit != ACE_player}) exitWith {
+        TRACE_1("null/dead/notplayer",_unit);
+        [_pfID] call CBA_fnc_removePerFrameHandler;
+    };
```
Comment (commy2) [inline addons/LegacyFatigue/functions/fnc_legacyFatigue.sqf]: `!alive objNull` is `true` now. And `objNull == objNull` is `false`. Something seems redundant.

**Comment by commy2** [comment]
Nice.

**Pull request closed without merging**